src/main/java/org/broadinstitute/hellbender/tools/exome/TargetArgumentCollection.java

```diff
@@ -104,6 +127,92 @@ public TargetArgumentCollection() {
         }
     }
 
+    /**
+     * Reads a target collection from a file and adds all the annotations in additional
+     * annotation files.
+     * @param targetFile the target collection file.
+     * @param targetAnnotations additional annotation files.
+     * @return never {@code null}.
+     */
+    private TargetCollection<Target> combineTargetAnnotations(final File targetFile,
+                                                              final List<File> targetAnnotations) {
+        final TargetCollection<Target> targets = readTargetCollection(targetFile);
+
+        final Map<Target, Map<TargetAnnotation, String>> annotationsByTarget =
+                composeTargetToAnnotationMap(targetAnnotations, targets);
+
+        final List<Target> fullyAnnotatedTargets = targets.targets().stream()
+                .map(originalTarget -> new Target(originalTarget.getName(),
+                                originalTarget.getInterval(),
+                                new HashTargetAnnotationCollection(annotationsByTarget.get(originalTarget))))
+                .collect(Collectors.toList());
+
+
+        return new HashedListTargetCollection<Target>(fullyAnnotatedTargets) {
+
+            @Override
+            public String name(final Target target) {
+                return target.getName();
+            }
+
+            @Override
+            public SimpleInterval location(final Target target) {
+                return target.getInterval();
+            }
+        };
+    }
+
+    /**
+     * Composes a Map from target to its annotations.
+     * <p>
+     *     The output map will contain exactly one entry per target in
+     *     the input target-collection {@code targets}. The value map
+     *     will contain all the annotations (keys) for each target along with their values.
+     *     Such maps may be empty if there is no annotation values for that target.
+     * </p>
+     *
+     * @param targetAnnotations files containing target annotations.
+     * @param targets the targets to annotate.
+     * @return never {@code null.}
+     */
+    private Map<Target, Map<TargetAnnotation, String>> composeTargetToAnnotationMap(final List<File> targetAnnotations,
```

Comment:
I think what you mean to review is the implementation type I use which is decided in composeTargetToAnnotationMap first code line. So instead of instantiate a HashMap instance implicitly thru Collectors.toMap I would need to make sure is uses a EnumMap.